base/namedtuple.jl

```diff
@@ -187,6 +187,26 @@ function map(f, nt::NamedTuple{names}, nts::NamedTuple...) where names
     NamedTuple{names}(map(f, map(Tuple, (nt, nts...))...))
 end
 
+@generated function _foldl_impl(
+    op,
+    init,
+    itr::Iterators.Pairs{<:Any,<:Any,<:Any,<:NamedTuple{names}},
+) where {names}
+    ex = :init
+    for (i, n) in enumerate(names)
+        ex = :(op($ex, $(QuoteNode(n)) => vals[$i]))
+    end
+    quote
+        @_inline_meta
+        vals = Tuple(itr.data)
+        $ex
+    end
```

Comment:
This should have an `if @generated` branch.

Reply:
I added it in https://github.com/JuliaLang/julia/pull/36291/commits/fcd2cbd1309c7a8cb06ae6eaf1e894f849f71cd2